File: backend/app/schemas/slide.py
"""Pydantic schemas for Slide API."""

from uuid import UUID

from pydantic import BaseModel


class SlideResponse(BaseModel):
    """Slide in API response."""

    id: UUID
    carousel_id: UUID
    order: int
    title: str
    body: str
    footer: str
    design_overrides: dict

    model_config = {"from_attributes": True}


class SlideUpdate(BaseModel):
    """Payload for PATCH: update title, body, footer."""

    title: str | None = None
    body: str | None = None
    footer: str | None = None

File: frontend/types/slide.ts
/** Single slide from GET /api/v1/carousels/{id}/slides. */
export interface SlideResponse {
  id: string;
  carousel_id: string;
  order: number;
  title: string;
  body: string;
  footer: string;
  design_overrides: Record<string, unknown>;
}

/** Body for PATCH /api/v1/carousels/{id}/slides/{slide_id}. */
export interface SlideUpdate {
  title?: string;
  body?: string;
  footer?: string;
}
